packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeProofFormatService.ts

```diff
@@ -0,0 +1,309 @@
+} from '../../../presentation-exchange/PresentationExchangeService'
+import type { W3cCredentialRecord, W3cVerifiablePresentation } from '../../../vc'
+import type { InputDescriptorToCredentials } from '../../models'
+import type { ProofFormatService } from '../ProofFormatService'
+import type {
+  ProofFormatCreateProposalOptions,
+  ProofFormatCreateReturn,
+  ProofFormatProcessOptions,
+  ProofFormatAcceptProposalOptions,
+  FormatCreateRequestOptions,
+  ProofFormatAcceptRequestOptions,
+  ProofFormatProcessPresentationOptions,
+  ProofFormatGetCredentialsForRequestOptions,
+  ProofFormatSelectCredentialsForRequestOptions,
+  ProofFormatAutoRespondProposalOptions,
+  ProofFormatAutoRespondRequestOptions,
+  ProofFormatAutoRespondPresentationOptions,
+} from '../ProofFormatServiceOptions'
+import type { PresentationSubmission } from '@sphereon/pex-models'
+
+import { Attachment, AttachmentData } from '../../../../decorators/attachment/Attachment'
+import { AriesFrameworkError } from '../../../../error'
+import { deepEquality } from '../../../../utils'
+import { PresentationExchangeService } from '../../../presentation-exchange/PresentationExchangeService'
+import { ProofFormatSpec } from '../../models'
+
+const PRESENTATION_EXCHANGE_PRESENTATION_PROPOSAL = 'dif/presentation-exchange/definitions@v1.0'
+const PRESENTATION_EXCHANGE_PRESENTATION_REQUEST = 'dif/presentation-exchange/definitions@v1.0'
+const PRESENTATION_EXCHANGE_PRESENTATION = 'dif/presentation-exchange/submission@v1.0'
+
+export class PresentationExchangeProofFormatService implements ProofFormatService<PresentationExchangeProofFormat> {
+  public readonly formatKey = 'presentationExchange' as const
+
+  private presentationExchangeService(agentContext: AgentContext) {
+    if (!agentContext.dependencyManager.isRegistered(PresentationExchangeService)) {
+      throw new AriesFrameworkError(
+        'PresentationExchangeService is not registered on the Agent. Please provide the PresentationExchangeModule as a module on the agent'
+      )
+    }
+
+    return agentContext.dependencyManager.resolve(PresentationExchangeService)
+  }
+
+  public supportsFormat(formatIdentifier: string): boolean {
+    return [
+      PRESENTATION_EXCHANGE_PRESENTATION_PROPOSAL,
+      PRESENTATION_EXCHANGE_PRESENTATION_REQUEST,
+      PRESENTATION_EXCHANGE_PRESENTATION,
+    ].includes(formatIdentifier)
+  }
+
+  public async createProposal(
+    agentContext: AgentContext,
+    { proofFormats, attachmentId }: ProofFormatCreateProposalOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const pexFormat = proofFormats.presentationExchange
+    if (!pexFormat) {
+      throw new AriesFrameworkError('Missing Presentation Exchange format in create proposal attachment format')
+    }
+
+    const { presentationDefinition } = pexFormat
+
+    ps?.validatePresentationDefinition(presentationDefinition)
```

Comment:
ps is always defined right (it should be as otherwise it could lead to unexpected behaviour cause validate is not called in that case)?

```suggestion
    ps.validatePresentationDefinition(presentationDefinition)
```